triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -345,29 +371,32 @@ object Server {
       case Failure(ex) => StartFailed(ex)
     }
 
-    // The server running server.
+    // The server running state.
     def running(binding: ServerBinding): Behavior[Message] =
       Behaviors
         .receiveMessage[Message] {
           case TriggerStarting(runningTrigger) =>
-            // Nothing to do at this time.
+            server.logTriggerStatus(runningTrigger, "starting")
             Behaviors.same
           case TriggerStarted(runningTrigger) =>
             // The trigger has successfully started. Update the
             // running triggers tables.
+            server.logTriggerStatus(runningTrigger, "running")
             server.addRunningTrigger(runningTrigger)
             Behaviors.same
           case TriggerInitializationFailure(runningTrigger, cause) =>
             // The trigger has failed to start. Send the runner a stop
             // message. There's no point in it remaining alive since
             // its child actor is stopped and won't be restarted.
+            server.logTriggerStatus(runningTrigger, "stopped: initialization failure")
             runningTrigger.runner ! TriggerRunner.Stop
             // No need to update the running triggers tables since
             // this trigger never made it there.
             Behaviors.same
           case TriggerRuntimeFailure(runningTrigger, cause) =>
             // The trigger has failed. Remove it from the running
             // triggers tables.
+            server.logTriggerStatus(runningTrigger, "stopped: runtime failure")
```

Comment:
It looks like your logs are a memory leak. You only ever write to them but never delete anything. This is probably not a huge issue atm but we should move them to postgres to avoid that.

Reply:
Yes, there is no attempt to control the memory here at this time. I hadn't considered having this data written to the DB - not against it.

Reply:
@rohanjr Thoughts?

Reply:
Makes sense (if the user supplies a db connection)